### ISSUE TITLE
getdns-1.0.0b2 compilation problems

### ISSUE THREAD

**Issue by simsong** [getdnsapi/getdns#207]
I had numerous problems attempting to compile getdns-1.0.0b2 on a new Centos7 system.
- getdns would not compile with version OpenSSL 1.0.2h installed in /usr/local/ssl. My base install was OpenSSL 1.0.1e-fips.
- When I removed /usr/local/ssl, getdns would not configure with the Centos libunbound. I had to install my own version of unbound (version 1.5.9).
- getdns requires that the unbound header unbound-event.h be installed. This header is not installed by default.

It would be useful if getdns releases were tested on clean VMs, and if getdns was distributed with a script that installed all of the necessary packages on a clean Centos 7 VM.

**Comment by wtoorop** [comment]
Hi Simon,

None of the issues you mention should have happened.  It should work
with openssl 1.02h (did you configure with --with-ssl=/usr/local/lib ?),
it should work with any libunbound and should not be dependent on the
availability of the unbound-event.h header.

Is it perhaps possible to show the errors you had?

We do test on clean VMs, but Centos 7 was not one of them (yet).
I'll try to add such an VM to our jekins setup and see what happens.

Thanks for the report!
-- Willem

Op 31-08-16 om 15:39 schreef Simson L. Garfinkel:

**Comment by simsong** [comment]
Thanks for the response.
I did not use --with-ssl=/usr/local/lib; I set -I and -L flags with LDFLAGS and CPPFLAGS. The configure file does check for unbound-event.h and I did get errors that a structure defined in that file was missing.

I can try going back to the stock libunbound and see what happens, but I won't be able to do it for a while.

I would like to use the version of getdns that's in the Centos7 repo, but I can't, as it has the DNSSEC CNAME validation bug that I had previously reported.

Simson
